phylanx/util/matrix_iterators.hpp

```diff
@@ -122,6 +134,15 @@ namespace phylanx { namespace util
         T* data_;
         std::size_t index_;
     };
+
+        template <typename T>
+        bool operator<(matrix_row_iterator<T> const& lhs, matrix_row_iterator<T> const& rhs)
+        {
+
+                return std::lexicographical_compare(lhs->begin(), lhs->end(),
+                                                    rhs->begin(), rhs->end());
+
+        }
```

Comment:
Hrmm, why do you think we need to overload `operator<()` for the iterators? Could you give me a rationale, please?

Reply:
I'm sorry the changes I made to matrix_iterator.hpp was what for my experiment I thought I had reverted the changes before the commit, this should not have been in the commit.